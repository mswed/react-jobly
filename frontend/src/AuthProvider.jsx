import React, { useState, useEffect } from 'react';
import JoblyApi from './api';

// We create the context
const AuthContext = React.createContext();

// We set up a component to manage the context
const AuthProvider = ({ children }) => {
  // We initialize the token with the value from localStorage
  const [token, setToken] = useState(localStorage.getItem('jobly-token'));

  // Whenever the token changes (login/logout) we need to update
  // both the api module and localStorage
  useEffect(() => {
    JoblyApi.token = token;
    localStorage.setItem('jobly-token', token || '');
  }, [token]);

  const login = async (username, password) => {
    try {
      const token = await JoblyApi.login(username, password);
      setToken(token);
      return { success: true };
    } catch (error) {
      return { success: false, error };
    }
  };

  const register = async (username, password, firstName, lastName, email) => {
    try {
      const token = await JoblyApi.register(username, password, firstName, lastName, email);
      setToken(token);
      return { success: true };
    } catch (error) {
      return { success: false, error };
    }
  };

  // Create the value object
  const value = { token, register };

  // Wrap everything in the context
  return <AuthContext.Provider value={value}>{children}</AuthContext.Provider>;
};

export { AuthContext, AuthProvider };
