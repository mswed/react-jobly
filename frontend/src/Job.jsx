import { Card, Row, Button } from 'react-bootstrap';
const Job = ({ title, company, salary, equity }) => {
  return (
    <Card className="mb-3">
      <Row className="g-0">
        <Card.Body>
          <Card.Title>{title}</Card.Title>
          {company && <Card.Text>{company}</Card.Text>}
          <ul>
            <li>Salary: {salary}</li>
            <li>Equity: {equity}</li>
          </ul>
          <div className="d-flex justify-content-end">
            <Button variant="warning">Apply</Button>
          </div>
        </Card.Body>
      </Row>
    </Card>
  );
};

export default Job;
