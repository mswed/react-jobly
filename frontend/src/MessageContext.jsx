import React, { useState } from 'react';

// We create the context
const MessageContext = React.createContext();

// We set up a component to manage the context
const MessagesProvider = ({ children }) => {
  const [message, setMessage] = useState(null);

  const showMessage = (text, variant = 'success', duration = 5000) => {
    setMessage({ text, variant, duration });
    // Auto clear
    if (duration) {
      setTimeout(() => {
        clearMessage();
      }, duration);
    }
  };

  const clearMessage = () => {
    setMessage(null);
  };

  return <MessageContext.Provider value={{ message, showMessage, clearMessage }}>{children}</MessageContext.Provider>;
};

export { MessageContext, MessagesProvider };
