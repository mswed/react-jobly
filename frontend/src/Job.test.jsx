import { it, expect } from 'vitest';
import { render } from '@testing-library/react';
import Job from './Job';
import { MemoryRouter } from 'react-router-dom';
import { AuthProvider } from './AuthProvider';
import { MessagesProvider } from './MessageContext';

it('renders without crashing', async () => {
  render(
    <MemoryRouter>
      <MessagesProvider>
        <AuthProvider>
          <Job />
        </AuthProvider>
      </MessagesProvider>
    </MemoryRouter>
  );
});

it('matches snapshot', async () => {
  const { asFragment } = render(
    <MemoryRouter>
      <MessagesProvider>
        <AuthProvider>
          <Job />
        </AuthProvider>
      </MessagesProvider>
    </MemoryRouter>
  );

  expect(asFragment()).toMatchSnapshot();
});
