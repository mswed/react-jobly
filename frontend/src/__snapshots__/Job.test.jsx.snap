// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="mb-3 card"
  >
    <div
      class="g-0 row"
    >
      <div
        class="card-body"
      >
        <div
          class="card-title h5"
        />
        <ul>
          <li>
            Salary: 
          </li>
          <li>
            Equity: 
          </li>
        </ul>
        <div
          class="d-flex justify-content-end"
        >
          <button
            class="btn btn-warning"
            type="button"
          >
            Apply
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
